### PR TITLE
Fix cache_utils for optimum.quanto kvcache quantization 

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do?
Supersedes https://github.com/huggingface/transformers/pull/34606 and updated a bit the docs

Quanto kv cache tests are passing

### CODE REVIEW

File: src/transformers/cache_utils.py [hunk 1]
```diff
@@ -781,6 +781,12 @@ def __init__(self, cache_config: CacheConfig) -> None:
         super().__init__(cache_config)
 
         if is_optimum_quanto_available():
+            optimum_quanto_version = version.parse(importlib.metadata.version("optimum-quanto"))
+            print(optimum_quanto_version)
```
Comment (dacorvo) [inline src/transformers/cache_utils.py:785]: I think you can remove that print statement.

**Review by dacorvo — APPROVED** [review]
Only a small nit

File: src/transformers/cache_utils.py (hunk 1 above)
Comment (SunMarc) [reply src/transformers/cache_utils.py:785]: oh yes my bad

**Review by zucchini-nlp — APPROVED** [review]
LGTM, thanks!

**Review by LysandreJik — APPROVED** [review]
Sounds good!

**Pull request merged**